sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/complexTypeCreator.scala

```diff
@@ -111,11 +110,12 @@ private [sql] object GenArrayData {
        """
       }
 
-      ("",
+      (s"$arrayName = new Object[$numElements];",
```

Comment:
we still create an object array every time here, which doesn't help with the GC. I think we can just create a local variable here.

Reply:
To make it local variable, I would like to use a new API at #19821.
After merging #19821, I will commit new version.